util.sh

```diff
@@ -0,0 +1,85 @@
+SLB_FILE=${SLB_FILE:=lz_header.bin}
+
+SL_SIZE=`hexdump "$SLB_FILE" -s2 -n2 -e '/2 "%u"'`
+
+validate_and_escape_hash () {
+	local TRIM=`echo -n "$1" | sed -r -e "s/ .*//"`
+	if (( ${#TRIM} != 64 && ${#TRIM} != 40 )); then
+		>&2 echo "\"$TRIM\" is not a valid SHA1/SHA256 hash"
+		return
+	fi
+	echo -n $TRIM | sed -r -e "s/([a-f0-9]{2})/\\\\\x\1/g"
+}
+
+# extend_sha* - extend initial PCR value (all 0s) with ${FILE}'s hash (SL only)
+# extend_sha* file.bin - extend initial PCR value with full file.bin's hash
+# extend_sha* HASH file.bin - extend PCR with value=HASH with file.bin's hash
+# extend_sha* HASH1 HASH2 - extend PCR with value=HASH1 with HASH2
+# extend_sha* HASH file file...
+# extend_sha* HASH HASH HASH... - as long as file is not first
+extend_sha1 () {
+	#echo $# >&2
+	local HASH1
+	local HASH2
+	if [ $# -eq 0 ]; then
```

Comment:
`case $#`, or does that do weird integer things?

Reply:
Done